src/cmd/services/m3dbnode/main/main_test.go

```diff
@@ -616,3 +634,42 @@ db:
                   endpoint: {{.InitialClusterEndpoint}}
 `
 )
+
```

Comment:
This is an extremely aggressive poll. Why no delay between polls? Will this introduce unwanted amounts of extra load? Not sure how expensive the read of shard set? I assume that it would be the `etcd` store.

Reply:
I don't think its aggressive. The context here is that it only happens when these integration tests are run, and they're all marked "big" which means that they'll always be run in sequence (never in parallel). So the poll is just aggressive to minimize humans waiting around for C.I to finish.